test/test_connectionpool.py

```diff
@@ -506,17 +507,43 @@ class _raise_once_make_request_function:
             successful response on subsequent calls.
             """
 
-            def __init__(self, ex: Type[BaseException]) -> None:
+            def __init__(
+                self, ex: Type[BaseException], pool: HTTPConnectionPool
+            ) -> None:
                 super().__init__()
                 self._ex: Optional[Type[BaseException]] = ex
-
-            def __call__(self, *args: Any, **kwargs: Any) -> httplib.HTTPResponse:
+                self._pool = pool
+
+            def __call__(
+                self,
+                conn: HTTPConnection,
+                method: str,
+                url: str,
+                *args: Any,
+                retries: Retry,
+                **kwargs: Any,
+            ) -> HTTPResponse:
                 if self._ex:
                     ex, self._ex = self._ex, None
                     raise ex()
-                response = httplib.HTTPResponse(MockSock)  # type: ignore[arg-type]
-                response.fp = MockChunkedEncodingResponse([b"f", b"o", b"o"])  # type: ignore[assignment]
-                response.headers = response.msg = httplib.HTTPMessage()
+                httplib_response = httplib.HTTPResponse(MockSock)  # type: ignore[arg-type]
+                httplib_response.fp = MockChunkedEncodingResponse([b"f", b"o", b"o"])  # type: ignore[assignment]
+                httplib_response.headers = httplib_response.msg = httplib.HTTPMessage()
+
+                response = HTTPResponse(
+                    body=httplib_response,
+                    headers=httplib_response.headers,  # type: ignore[arg-type]
+                    status=httplib_response.status,
+                    version=httplib_response.version,
+                    reason=httplib_response.reason,
+                    original_response=httplib_response,
+                    retries=retries,
+                    request_method=method,
+                    request_url=url,
+                    preload_content=False,
+                    connection=conn,
+                    pool=self._pool,
+                )
```

Comment:
I think I messed up here, I should be getting `response_conn` from `kwargs` and using that to construct my `HTTPResponse` object. Since that's how its currently done in `_make_request()` and `getresponse()`.

Reply:
done